src/common/CryptoUtils.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import forge from 'node-forge';
+import { Messages } from '@salesforce/core';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+
+const messages = Messages.loadMessages('@salesforce/lwc-dev-mobile-core', 'crypto-utils');
+
+export type PEMCertificate = {
+    derCertificate: string;
+    pemCertificate: string;
+    pemPrivateKey: string;
+    pemPublicKey: string;
+};
```

Comment:
NOTE: iOS and Android need cert in DER format

I've been debating whether only have a generic `certificate: string` field here rather than both `pem` and `der` certificates.

If we choose to have a generic field then we need to to add another param to `generateSelfSignedCert` to indicate whether we want PEM or DER format + add extra code logic around using the param to generate the rightly formatted cert.

But in my experience so far in using `node-forge` it seems to be really fast. Bulk of execution time is taken by `createCertificate()` and `sign()` methods of `node-forge` (which still is under 1 sec).... once the signed cert is created, converting it to PEM or DER seems to be super fast.

Because of the fast speed of `node-forge` I oped to generate both PEM and DER formats of the cert at the same time and let the caller consume whichever it needs.

If we don't like this then I can update to only have a generic `certificate: string` field here and pass an extra param to `generateSelfSignedCert`

Reply:
I have a few thoughts on this:

- I think it's fine to represent both formats in the same data structure.
- I think we should call the data structure itself something different than `PEMCertificate` in that case, like `SSLCertificateData` or somesuch.
- I don't know about Android, but I am able to drag and drop PEM certificates into iOS simulators and have that "just work".
- `derCertificate` should likely not be represented as a `string`, as it's a binary data structure to represent the certificate.

Reply:
I just finished testing on Android (API 24 - 34) and it works with both PEM and DER.

I was unable to use PEM on iOS via drag/drop the PEM file generated by the code in here. My simulator gives the error `Simulator device failed to complete the requested operation`

But instead of drag/drop if I open safari on mobile and then navigate to `file://path/to/cert` then it installs the cert just fine. And since I was unable to find a way to auto-install the cert on iOS, we'll have to go the manual route which means using mobile safari so I'll go ahead and remove the DER support from this PR.

Reply:
Ah I see. My functioning drag-and-drop PEM for iOS was for a certificate created with `openssl`, so there are probably some differences between that and this process. I'm good with whichever approach we take that makes it less clicks or typing for the user.

Reply:
> derCertificate should likely not be represented as a string, as it's a binary data structure to represent the certificate.

It actually is. We generate the DER cert using this line:
```
const derCert = forge.asn1.toDer(forge.pki.certificateToAsn1(cert)).getBytes();
```

The call to `getBytes()` actually returns a string but when we are saving the content to file we should just use `fs.writeFileSync(path, derCertificate, { encoding: 'binary' })`

Reply:
Oh sorry, I didn't follow the definition through—just going on what I knew about the format.